peas/weather.py

```diff
@@ -816,111 +754,42 @@ def make_safety_decision(self, current_values):
-            elif last_cloud > threshold_cloudy:
-                cloud_condition = 'Cloudy'
-            else:
-                cloud_condition = 'Clear'
-            self.logger.debug('Cloud Condition: {} (Sky-Amb={:.1f} C)'.format(cloud_condition, sky_diff[-1]))
+        super._get_cloud_safety()
 
         return cloud_condition, sky_safe
 
     def _get_wind_safety(self, current_values):
-        safety_delay = self.safety_delay
         entries = self.weather_entries
 
-        end_time = dt.utcnow()
-
-        threshold_windy = self.cfg.get('threshold_windy', 20.)
-        threshold_very_windy = self.cfg.get('threshold_very_windy', 30)
-
-        threshold_gusty = self.cfg.get('threshold_gusty', 40.)
-        threshold_very_gusty = self.cfg.get('threshold_very_gusty', 50.)
-
         # Wind (average and gusts)
         wind_speed = [x['wind_speed_KPH']
                       for x in entries
                       if 'wind_speed_KPH' in x.keys()]
 
-        if len(wind_speed) == 0:
-            self.logger.debug('  UNSAFE: no wind speed readings found')
-            wind_safe = False
-            gust_safe = False
-            wind_condition = 'Unknown'
-            gust_condition = 'Unknown'
-        else:
-            start_time = entries[0]['date']
-            if type(start_time) == str:
-                start_time = date_parser(entries[0]['date'])
-
-            typical_data_interval = (end_time - start_time).total_seconds() / len(entries)
-
-            mavg_count = int(np.ceil(120. / typical_data_interval))  # What is this 120?
-            wind_mavg = movingaverage(wind_speed, mavg_count)
-
-            # Windy?
-            if max(wind_mavg) > threshold_very_windy:
-                self.logger.debug('  UNSAFE:  Very windy in last {:.0f} min. Max wind speed {:.1f} kph'.format(
-                    safety_delay, max(wind_mavg)))
-                wind_safe = False
-            else:
-                wind_safe = True
+        wind_gust = max(wind_speed)
 
-            if wind_mavg[-1] > threshold_very_windy:
-                wind_condition = 'Very Windy'
-            elif wind_mavg[-1] > threshold_windy:
-                wind_condition = 'Windy'
-            else:
-                wind_condition = 'Calm'
-            self.logger.debug('  Wind Condition: {} ({:.1f} km/h)'.format(wind_condition, wind_mavg[-1]))
-
-            # Gusty?
-            if max(wind_speed) > threshold_very_gusty:
-                self.logger.debug('  UNSAFE:  Very gusty in last {:.0f} min. Max gust speed {:.1f} kph'.format(
-                    safety_delay, max(wind_speed)))
-                gust_safe = False
-            else:
-                gust_safe = True
+        end_time = dt.utcnow()
+        start_time = date_parser(entries[0]['date'])
+        typical_data_interval = (end_time - start_time).total_seconds() / len(entries)
+        mavg_count = int(np.ceil(120. / typical_data_interval))  # What is this 120?
```

Comment:
Good question...

Reply:
hmmm ... this one is probably on me to answer as the original author way back when.  At a glance, I think what is happening is I want to get ~2 minutes of wind data to go in to the moving average for the wind.  This just figures out how many data points to average in the moving average to get ~2 minutes (120 seconds).  The ceil just forces it to always round up.

Reply:
Thanks @joshwalawender  FYI, this will probably be closed and we will merge into `AstroHuntsman/POCS` and then do a PR from there to here when done.

@jermainegug While you are at it can you turn this into a variable up above, i.e. `moving_avg_seconds = 120`